guardian/prosemirror-elements
# feat: Convert video element to 'standard' element

## What does this change?

Renames the `video` element to a `standard` element, and adds the supported types to the relevant data transformer.

A standard element represents every element covered by https://github.com/guardian/flexible-model/blob/f9d30ad2bb19446a9226ab1bd8418b4aaa03d762/src/main/thrift/content.thrift#L696.
 
In practice, it also covers the legacy (non-content-atom) audio and video elements, which only use a subset of their flexible-model fields.

This PR makes one modification to the element fields – every text-based element field is removed if empty, to match the current element behaviour.

Per the description in flexible-model, the data requirements for this element also match the `table` element, but that has a diferent presentation, and so we'll add it in a separate PR.

## How to test

It's worth testing with a variety of elements of different types, from different periods, in our consuming CMS' CODE environment. I've included a sample of pieces from that env, ordered by creation date, to check.

This PR contains fixture tests for each element type per #213, which offer some assurance that our new element type will not affect existing elements.

<details>
<summary>Pieces with video elements (these should be unchanged)</summary>
617805ee8f08f18a166abde6,2021-10-26
61d4740d8f08f06c8872433b,2022-01-04
6013f63a8f08be3f4348fd83,2021-01-29
5b8959e3e4b05cd0bedf4a7b,2018-08-31
610969698f088b0b4e9fdcd8,2021-08-03
619682ab8f08664e0af85c89,2021-11-18
5fc617948f086ce9248f3761,2020-12-01
5ff6e97f8f08c7a368e6e4ae,2021-01-07
600800b28f0850df55867fa4,2021-01-20
60ee4bb28f08331a6b2864e4,2021-07-14
60ee4bb28f08331a6b2864e4,2021-07-14
601a859b8f0837302b86d97c,2021-02-03
60cb61488f081fcfa2d35b01,2021-06-17
60cb61488f081fcfa2d35b01,2021-06-17
611bc8008f085022754cbe31,2021-08-17
60420bfc8f08e95b70aead0c,2021-03-05
6059c7cf8f083158d979fecf,2021-03-23
606dade58f081a7273c4dea2,2021-04-07
607d57568f08b5689bedf1e1,2021-04-19
5ad6f5eae4b04e8efbbb7457,2018-04-18
</details>

<img width="826" alt="Screenshot 2022-03-01 at 09 04 04" src="https://user-images.githubusercontent.com/7767575/156138534-ff3af05e-890a-4778-86e2-09607e33595d.png">

<details>
<summary>Pieces with audio elements</summary>
60cb61488f081fcfa2d35b01,2021-06-17
603f69ec8f08c4b040ad3bde,2021-03-03
5f9add148f08e8dfdcc50fab,2020-10-29
5c88cac9e4b078b9d50aa516,2019-03-13
5b55a695e4b099be0ba37258,2018-07-23
5a844a86e4b0c73a428405e5,2018-02-14
5a69e1abe4b00f225a1c7aeb,2018-01-25
5a69ccfee4b0473d0bc10221,2018-01-25
5a6863a8e4b08c0147181fb2,2018-01-24
5a5ce770e4b046e0befbeae7,2018-01-15
5a5c9a62e4b0fe5e06bfa569,2018-01-15
5a5a48ffe4b0fe5e06bf9923,2018-01-13
5a55ffaae4b043ead794c45a,2018-01-10
5a4d0257e4b05efedfbd6fe0,2018-01-03
5a4bbd0ce4b01254e1d211b9,2018-01-02
5a4379b1b7f7b10656af30d9,2017-12-27
5a3bb77a7264a8077e24e74a,2017-12-21
5a3a9ffab428d606a9d9c8f9,2017-12-20
5a3a6bdfc7507c076d1334f6,2017-12-20
5a389bb0bbf0e7073eb10b2a,2017-12-19
</details>

<img width="890" alt="Screenshot 2022-03-01 at 08 56 30" src="https://user-images.githubusercontent.com/7767575/156138208-892744d4-4402-4b6c-b30f-e1b6b9578957.png">

<details>
<summary>Pieces with map elements</summary>
60cb61488f081fcfa2d35b01,2021-06-17
5ce68b988f08f34f7281b941,2019-05-23
59c173c2e4b0f6beafd39d88,2017-09-19
59bcde75e4b06eeee7d810f8,2017-09-16
59471af4e4b014a733d638a5,2017-06-19
58ba0f6ee4b09c1763ef5eda,2017-03-04
58b5da25e4b030b6f7c901d6,2017-02-28
58ab6b8be4b030b6f7c8d717,2017-02-20
587fbbd8e4b01065cf597f03,2017-01-18
57ff8362e4b01ffb8ffcb4a4,2016-10-13
57dff371e4b0ac5fd0e5dfbf,2016-09-19
577094aee4b0f43038109f93,2016-06-27
57638603e4b088b489cb2488,2016-06-17
57619448e4b096841ef399b9,2016-06-15
5760c5f4e4b005ab05d211aa,2016-06-15
5760b806e4b0505cd1bcbc7c,2016-06-15
56dd68a1e4b0a02aed206487,2016-03-07
568331dae4b0629150aaf589,2015-12-30
567426b7e4b0fcd39b5714e2,2015-12-18
560e9348e4b06bd936f9c754,2015-10-02
</details>

<img width="955" alt="Screenshot 2022-03-01 at 09 02 03" src="https://user-images.githubusercontent.com/7767575/156138233-57ae0c12-b859-418e-be85-00c188294ce5.png">

<details>
<summary>Pieces with document elements</summary>
60cb61488f081fcfa2d35b01,2021-06-17
6037b00b8f08d29f4c26eb46,2021-02-25
5aa2a3fbe4b00a04356fb5ef,2018-03-09
5a9e7be0e4b0f55650b88a7c,2018-03-06
5a232f930470380699b5386e,2017-12-02
59f74a62a885b806b2a4e7f6,2017-10-30
5951a06ce4b0cdea6d8657e4,2017-06-27
59160506e4b0a8ea08b6d9f9,2017-05-12
575915e5e4b01a5ff948b084,2016-06-09
57556bb8e4b0be576f9d9b55,2016-06-06
573c780ee4b04a0378340cf3,2016-05-18
57220c38e4b02046f02ebdc4,2016-04-28
56fa2eabe4b060534dcd586d,2016-03-29
56f1786fe4b0d8d383385a86,2016-03-22
56b3af27e4b05b93c48cd27b,2016-02-04
566bf8cae4b0c8f491ad28f9,2015-12-12
56588e65e4b03bf401a0bc0d,2015-11-27
56583dede4b0dd35d29c32eb,2015-11-27
559e8119e4b038c9e1db1b25,2015-07-09
554b4494e4b0c7299d72c995,2015-05-07
</details>

(Unsure how useful that preview is! 🤔)

<img width="955" alt="Screenshot 2022-03-01 at 09 01 54" src="https://user-images.githubusercontent.com/7767575/156138251-f8e30355-0a1c-4664-b915-449e13751c1e.png">

Tested in local Composer with all of the above, including altering captions and mandatory flags for each.

## How can we measure success?

A map, audio, and document element that works with the data we currently have. A video element that continues to work as expected.

## Accessibility
